app/controllers/stripe-setup/responsible-person/post.controller.js

```diff
@@ -106,17 +106,17 @@ module.exports = (req, res) => {
     pageData['errors'] = errors
     return response(req, res, 'stripe-setup/responsible-person/index', pageData)
   } else if (lodash.get(req.body, 'answers-checked') === 'true') {
-    return createPerson(res.locals.stripeAccount.stripeAccountId, buildStripePerson(formFields))
-      .then(() => {
-        return connector.setStripeAccountSetupFlag(req.account.gateway_account_id, 'responsible_person', req.correlationId)
-      })
-      .then(() => {
-        return res.redirect(303, paths.dashboard.index)
-      })
-      .catch(error => {
-        logger.error(`[requestId=${req.correlationId}] Error creating responsible person with Stripe - ${error.message}`)
-        return renderErrorView(req, res, 'Please try again or contact support team')
-      })
+    try {
+      const stripeAccountId = res.locals.stripeAccount.stripeAccountId
+      const persons = await listPersons(stripeAccountId)
+      const person = persons.pop()
+      await updatePerson(stripeAccountId, person.id, buildStripePerson(formFields))
+      await connector.setStripeAccountSetupFlag(req.account.gateway_account_id, 'responsible_person', req.correlationId)
+      await res.redirect(303, paths.dashboard.index)
```

Comment:
`await` is much more _promising_ than `then`

Reply:
🤦‍♂️